### PR TITLE
Feature/homology annotation - Dumping, Compressing and Syncing the homologies 

### PR DESCRIPTION
**Pull request by Simarpreet-Kaur-Bhurji**
## Description
Adding a step to the homology pipeline to dump, compress and sync the homologies.

A runnable called DumpSpeciesDBToTsv.pm was created, this runnable takes a compara registry alias to output the corresponding database name.
And gets the required parameters and runs the python script to dump homologies to a tsv file. 

The steps to integrate these changes into the homology pipeline were added to the PerSpeciesCopyFactory.pm file.

Corresponding changes to the pipeline wide parameters and output directory were made in the HomologyAnnotation_conf.pm file.

**Related JIRA tickets:**
- [ENSCOMPARASW-6050](https://www.ebi.ac.uk/panda/jira/browse/ENSCOMPARASW-6050)

## Overview of changes
A step to dump, compress and sync the homologies was added to the homology pipeline. 

## Testing
_How was this tested? Have new unit tests been included?_
The dumping, compressing and syncing was tested by running the pipeline for a single and multiple species.

---

## PR review checklist

- [ ] Is the PR against an appropriate branch?
- [ ] Does the code adhere to coding guidelines?
- [ ] Does the code do what it claims to do?
- [ ] Is the code readable? Is it appropriately documented?
- [ ] Is the logic in the correct place?
- [ ] Was the code tested appropriately? Are there unit tests? Are unit tests self-contained and non-redundant?
- [ ] Did Travis CI pass for the code in the PR? Is Codecov acceptable based on the included/updated unit tests?
- [ ] Will the new code fail gracefully?
- [ ] Does the code follow good practice for writing performant code (e.g. using a database transaction rather than repeated queries outside of a transaction)?
- [ ] Does it bring in an unnecessary dependency?
- [ ] If you are reviewing a new analysis, is it future-proof and pluggable?
- [ ] Does the PR meet agile guidelines?

### CODE REVIEW

File: scripts/dumps/dump_homologies.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+parser.add_argument(
+    '-o', metavar='output', type=str, help="Output tsv.", required=True)
+
+
+if __name__ == '__main__':
+    args = parser.parse_args()
+
+    ref_db = args.r
+    species_db_url = args.u
+    server_url, species_db = species_db_url.rsplit("/", 1)
```
Comment (twalsh-ebi) [inline scripts/dumps/dump_homologies.py:45]:
```suggestion
```

**Review by marcoooo — APPROVED** [review]

File: modules/Bio/EnsEMBL/Compara/PipeConfig/Parts/PerSpeciesCopyFactory.pm [hunk 2]
```diff
@@ -74,10 +74,70 @@ sub pipeline_analyses_create_and_copy_per_species_db {
...
+                '2' => [{'GenomeDirectoryPaths'=> INPUT_PLUS()}],
+            }
+        },
+
+        {   -logic_name        => 'GenomeDirectoryPaths',
+            -module            => 'Bio::EnsEMBL::Production::Pipeline::FileDump::DirectoryPaths',
+            -max_retry_count   => 1,
+            -analysis_capacity => 20,
+            -parameters        => {
+
```
Comment (twalsh-ebi) [inline modules/Bio/EnsEMBL/Compara/PipeConfig/Parts/PerSpeciesCopyFactory.pm:98]:
```suggestion
```

File: scripts/dumps/dump_homologies.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+Dump rapid release homologies into TSV files.
+Example::
+    $ python dump_homologies.py -u \
+            "mysql://ensro@mysql-ens-compara-prod-2:4522/accipiter_gentilis_compara_105" \
+            -r ensembl_compara_references -o test.tsv
+"""
+
+import argparse
+import csv
+from sqlalchemy import create_engine, text
```
Comment (twalsh-ebi) [inline scripts/dumps/dump_homologies.py:28]:
```suggestion

from sqlalchemy import create_engine, text
```

**Comment by twalsh-ebi** [comment]
Merging this PR as the Travis failures are unrelated.

**Pull request merged**